pkg/network/dns_snooper_test.go

```diff
@@ -264,7 +264,9 @@ func getStats(
 	snooper *SocketFilterSnooper,
 	expectedCount int,
 ) map[dnsKey]map[string]dnsStats {
```

Comment:
This looks like a good candidate for [`require.Eventually`](https://godoc.org/github.com/stretchr/testify/require#Eventually)

Reply:
There are cases where the loop is expected to break because of the timeout, not the main condition being true. I guess we can make the timeout itself another condition but that would defeat the purpose somewhat.

Reply:
Does the `waitFor` parameter to `require.Eventually` not satisfy that the timeout requirement?

Reply:
If the condition does not return `True` in `waitFor` time, then an error gets thrown - resulting in the Test to fail.